language/vm/vm_runtime/src/txn_executor.rs

```diff
@@ -251,31 +254,44 @@ where
                     if callee_function_ref.is_native() {
                         let module_name: &str = callee_function_ref.module().name();
                         let function_name: &str = callee_function_ref.name();
-                        let native_return = dispatch_native_call(
-                            &mut self.execution_stack,
-                            module_name,
-                            function_name,
-                        )
-                        .map_err(|_| VMInvariantViolation::LinkerError)?;
-                        try_runtime!(self.gas_meter.consume_gas(
-                            GasUnits::new(native_return.cost()),
-                            &self.execution_stack
-                        ));
-                        match native_return.get_return_value() {
-                            NativeReturnType::ByteArray(value) => {
-                                self.execution_stack.push(Local::bytearray(value));
-                                // Call stack is not reconstructed for a native call, so we just
-                                // proceed on to next instruction.
-                            }
-                            NativeReturnType::Bool(value) => {
-                                self.execution_stack.push(Local::bool(value));
-                                // Call stack is not reconstructed for a native call, so we just
-                                // proceed on to next instruction.
-                            }
-                            NativeReturnType::U64(value) => {
-                                self.execution_stack.push(Local::u64(value));
-                                // Call stack is not reconstructed for a native call, so we just
-                                // proceed on to next instruction.
+                        if module_name == EVENT_NAME && function_name == EMIT_EVENT_NAME {
+                            let msg = try_runtime!(self.execution_stack.pop_as::<ByteArray>());
+                            let count = try_runtime!(self.execution_stack.pop_as::<u64>());
+                            let key = try_runtime!(self.execution_stack.pop_as::<ByteArray>());
+                            let guid = AccountAddress::try_from(key.as_bytes())
+                                .map_err(|_| VMInvariantViolation::EventKeyMismatch)?;
+                            self.event_data.push(ContractEvent::new(
+                                AccessPath::new(guid, vec![]),
```

Comment:
Shouldn't we change the type of `AccessPath` to `guid` in `ContractEvent`.

Reply:
Yes. I was intended to clean it up in another PR to keep this PR small. Will add TODOs